connectors/src/connectors/zendesk/temporal/workflows.ts

```diff
@@ -404,18 +394,31 @@ async function runZendeskBrandTicketsSyncActivities({
   currentSyncDateMs: number;
   forceResync: boolean;
 }) {
-  let cursor: string | null = null;
-  let hasMore = true;
-
-  while (hasMore) {
-    const result = await syncZendeskTicketsBatchActivity({
+  await runWithPagination((cursor) =>
+    syncZendeskTicketBatchActivity({
       connectorId,
       brandId,
       currentSyncDateMs,
       forceResync,
       cursor,
-    });
+    })
+  );
+}
+
+/**
+ * Runs an activity function with cursor-based pagination.
+ */
+async function runWithPagination(
```

Comment:
Let's call it `runZendeskActivityWithPagination`

Name is too generic for what it actually does (returns hasMore and cursor, which are zendesk specific)

Reply:
Will do 👍